### DISCUSSION TITLE
Question about - Exchange OAuth 1.0 Token for OAuth 2.0 Token

### DISCUSSION THREAD

**Discussion by Makebuildmodify** [etsy/open-api#792] [Q&A]
Hi, I'm reading the documentation for exchanging OAuth 1.0 tokens for OAuth 2.0 tokens and was wondering if I make an exchange will my OAuth 1.0 token still be valid? 

I would like to keep using my current API v2 script to list my listings until I get the API v3 version coded up. So I need both types of tokens. 

The wording in the documentation is not clear as to what happens to the OAuth 1.0 token after the exchange.

https://developers.etsy.com/documentation/essentials/authentication/#exchange-oauth-10-token-for-oauth-20-token

**Comment by baahar** [comment]
Hi,

In my experience when I request an OAuth 2.0 token for a shop I can still use API v2 endpoints with it.

**Comment by Makebuildmodify** [comment]
@baahar, Thanks!

Just to make sure I'm understand you correctly, I should be able to use my API v2 script with Oauth 2.0 tokens? Which means that if I do a token exchange and the 0Auth 1.0 token no longer works I should still be able to run my old script, right?

**Reply by baahar** [reply]
Yes. That was my experience. I started re-programming my access functions one by one, so some were accessing the old endpoints and some the new. I have not checked whether all of them behaved as before, but I didn't get any error messages.

But to make sure, I would first try migrating the token of one shop and check if all your critical scripts still work: 
https://developers.etsy.com/documentation/essentials/authentication#exchange-oauth-10-token-for-oauth-20-token

**Comment by Makebuildmodify** [comment]
@baahar Thanks!

That's great to here! I never imagined that I could run to both endpoints from single script. But it totally makes sense because each request is pretty much script agnostic.  Very helpful, thank you!

**Comment by Silviapso** [comment]
@baahar can you please give us more details on how you used an Oauth 2.0 token to call Etsy Api v2? Did you use an access token or the refresh token?

**Reply by coder771** [reply]
@Silviapso did you find out if access token or refresh token is needed. Also what do we need to do once access token Token expires after the exchange token. Do we have to call Exchange token method each time when access token expires

**Reply by Silviapso** [reply]
@coder771 we did not find how solve it. Anyway in the meantime V2 is not available anymore

**Comment by haihoang99** [comment]
Someone please show me how to get the Legacy Token

**Reply by Makebuildmodify** [reply]
Are you talking about the API v2 token? If you are I'm pretty sure that API is no longer in use.

**Reply by haihoang99** [reply]
I mean this API: https://developers.etsy.com/documentation/essentials/authentication/#exchange-oauth-10-token-for-oauth-20-token, idk how to get legacy token from where ? can u help me pls

**Reply by Makebuildmodify** [reply]
@haihoang99 Do you have an OAuth 1.0 token from your old app?

**Reply by haihoang99** [reply]
yes, i used oauth token 1.0 for legacy token, but it not work

**Reply by Makebuildmodify** [reply]
@haihoang99 

You can use the command line tool 'Curl' to do this. 

This is what is should look like: 

`curl -d 'grant_type=token_exchange&client_id=<YOUR OLD KEYSTRING HERE>&legacy_token=<YOUR OLD LEGACY TOKEN HERE>' https://api.etsy.com/v3/public/oauth/token`

`Replace the:  "<YOUR OLD KEYSTRING HERE>" and the  "<YOUR OLD LEGACY TOKEN HERE>"  with your information.`

The Keystring can be found here: https://www.etsy.com/developers/your-apps   (You have to be logged into the correct shop) 

The Legacy token is the token that you should have from the previous app.

Here is a good page for Curl if you need it: https://everything.curl.dev/http/post/simple

**Reply by Makebuildmodify** [reply]
@haihoang99  How did that work for you?

**Reply by haihoang99** [reply]
im try but it's not work, im have a request to connect my store with old api and etsy return me token but i cant used that token for this api

**Reply by Makebuildmodify** [reply]
@haihoang99  did you try the code that I gave you?

**Comment by Gareth064** [comment]
Part of me believes that the Exchange Auth1 for Auth2 endpoint no longer works since they also remove the old API now.

Best to email etsy devs at developer@etsy.com and ask, in case it has been deprecated and they just haven't updated their documentation.

I also have a memory (but can't find the proof) that they said you have a certain amount of time to use the exchange. That might have no expired.

**Reply by Makebuildmodify** [reply]
@Gareth064

Interesting, I tried the exchange with my old app keys and managed to get a response with what looks like a well formed token set. Although, I didn't try to use it because I already have everything working for my new app. I only ran the test to see if the exchange was still working. I just assumed that if I received a token set that is would be useable.
